bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/LivisiWebSocket.java

```diff
@@ -0,0 +1,159 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LivisiWebSocket} implements the websocket for receiving constant updates
+ * from the LIVISI SmartHome web service.
+ *
+ * @author Oliver Kuhl - Initial contribution
+ * @author Sven Strohschein - Renamed from Innogy to Livisi
+ */
+@NonNullByDefault
+@WebSocket
+public class LivisiWebSocket {
+
+    private final Logger logger = LoggerFactory.getLogger(LivisiWebSocket.class);
+
+    private final HttpClient httpClient;
+    private final EventListener eventListener;
+    private final URI webSocketURI;
+    private final int maxIdleTimeout;
+
+    private @Nullable Session session;
+    private @Nullable WebSocketClient client;
+    private boolean closing;
+
+    /**
+     * Constructs the {@link LivisiWebSocket}.
+     *
+     * @param eventListener the responsible
+     *            {@link org.openhab.binding.livisismarthome.internal.handler.LivisiBridgeHandler}
+     * @param webSocketURI the {@link URI} of the websocket endpoint
+     * @param maxIdleTimeout max idle timeout
+     */
+    public LivisiWebSocket(HttpClient httpClient, EventListener eventListener, URI webSocketURI, int maxIdleTimeout) {
+        this.httpClient = httpClient;
+        this.eventListener = eventListener;
+        this.webSocketURI = webSocketURI;
+        this.maxIdleTimeout = maxIdleTimeout;
+    }
+
+    /**
+     * Starts the {@link LivisiWebSocket}.
+     */
+    public synchronized void start() throws Exception {
+        if (client == null || client.isStopped()) {
+            client = startWebSocketClient();
+        }
+
+        if (session != null) {
+            session.close();
+        }
+
+        logger.debug("Connecting to LIVISI SmartHome webSocket...");
+        session = client.connect(this, webSocketURI).get();
+    }
+
+    /**
+     * Stops the {@link LivisiWebSocket}.
+     */
+    public synchronized void stop() {
+        this.closing = true;
+        if (isRunning()) {
+            logger.debug("Closing session...");
+            session.close();
+            session = null;
+        } else {
+            session = null;
+            logger.trace("Stopping websocket ignored - was not running.");
+        }
+        if (client != null) {
+            try {
+                client.stop();
+                client.destroy();
+            } catch (Exception e) {
+                logger.debug("Stopping websocket failed", e);
+            }
+            client = null;
+        }
+    }
+
+    /**
+     * Return true, if the websocket is running.
+     *
+     * @return true if the websocket is running, otherwise false
+     */
+    public synchronized boolean isRunning() {
+        return session != null && session.isOpen();
+    }
+
+    @OnWebSocketConnect
+    public void onConnect(Session session) {
+        this.closing = false;
+        logger.info("Connected to LIVISI SmartHome webservice.");
```

Comment:
The log level can be reduced to **debug**. For reference: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
solved